.github/workflows/DiffTool-CI.yml

```diff
@@ -0,0 +1,149 @@
+name: diff-java-tool PR checks
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  # Allows manual triggering test changes in checkstyle/test-configs
+  workflow_dispatch:  
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    env:
+      JAVA_VERSION: '21'
+      CONFIG_DIR: '.ci-config'
+      DIFFTOOL_JAR: "diff-java-tool.jar"
+      PATCH_DIFF_TOOL_VERSION: "0.1-SNAPSHOT"
+      CI_TEMP_DIR: '.ci-temp'
+
+    steps:
+      - name: Create CI Temp Directory
+        run: mkdir -p "${{ env.CI_TEMP_DIR }}"
+
+      - name: Checkout Main Repository into CI Temp Directory
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          path: ${{ env.CI_TEMP_DIR }}/test-configs
```

Comment:
main repo should just be cloned to be main folder where all operation is happening, all other downloads/clones are in .ci-temp folders to not damage `git status` of test-configs

Reply:
done